x-pack/legacy/plugins/siem/public/components/ml/api/anomalies_table_data.ts

```diff
@@ -22,17 +22,18 @@ export interface Body {
 }
 
 export const anomaliesTableData = async (body: Body, signal: AbortSignal): Promise<Anomalies> => {
-  const response = await fetch(`${chrome.getBasePath()}/api/ml/results/anomalies_table_data`, {
-    method: 'POST',
-    credentials: 'same-origin',
-    body: JSON.stringify(body),
-    headers: {
-      'content-Type': 'application/json',
-      'kbn-system-api': 'true',
-      'kbn-xsrf': 'true',
-    },
-    signal,
-  });
-  await throwIfNotOk(response);
-  return response.json();
+  const response = await npStart.core.http.fetch<Anomalies>(
+    '/api/ml/results/anomalies_table_data',
+    {
+      method: 'POST',
+      credentials: 'same-origin',
+      body: JSON.stringify(body),
+      headers: { 'kbn-system-api': 'true' },
+      asResponse: true,
+      signal,
+    }
+  );
+
+  await throwIfNotOk(response.response!);
```

Comment:
Speaking of `throwIfNotOk`, I wonder if we should expand it to take `Response | undefined`, as the `response.response!` can be undefined if it errors out during some part of the request (not sure where/when though?). Flow should be fine as it'll still throw from within that function, but probably better to explicitly handle the `undefined`.

https://github.com/elastic/kibana/blob/4980387c17be00fc91aef591ddcb085945c3a18e/src/core/public/http/types.ts#L246-L247

Reply:
Yeah, I was unable to reproduce that behavior myself: any 4xx or 5xx resulted in an error being thrown. Regardless: 221eb62dd